include/seqan/bam_io/bam_file.h

```diff
@@ -248,10 +248,10 @@ readRecords(TRecords & records, FormattedFile<Bam, Input, TSpec> & file, TSize m
 {
     String<CharString> & buffers = context(file).buffers;
     if ((TSize)length(buffers) < maxRecords)
-    {
         resize(buffers, maxRecords, Exact());
+    if ((TSize)length(records) < maxRecords)
```

Comment:
Could you please use static_cast according to our style guide http://seqan.readthedocs.org/en/latest/StyleGuide/Cpp.html#static-cast. 
It would be great if you could change it above in line 250 as well.
:+1: